src/relayclient/ClientWinstonLogger.ts

```diff
@@ -0,0 +1,63 @@
+import Cookies from 'js-cookie'
+import log from 'loglevel'
+import winston, { transport } from 'winston'
+import { HttpTransportOptions } from 'winston/lib/winston/transports'
+
+import { NpmLogLevel } from './types/Aliases'
+import { gsnRuntimeVersion } from '../common/Version'
+import { LoggerInterface } from '../common/LoggerInterface'
+
+const format = winston.format.combine(
+  winston.format.uncolorize(),
+  winston.format.timestamp(),
+  winston.format.simple()
+)
+
+const service = 'gsn-client'
+const userIdKey = 'gsn-client-user-id'
+
+const isBrowser = typeof window !== 'undefined'
+
+function createUserId (): string {
+  const userId = `${userIdKey}${Date.now()}`
+  Cookies.set(userIdKey, userId)
+  return userId
+}
+
+export function createLogger (level: NpmLogLevel, customerToken: string, hostOverride: string, userIdOverride: string): LoggerInterface {
```

Comment:
- use named params, ({})
- userIdOverride is very useful (and can be named userId). others are less useful to end-users
- need setLevel to change active logger, with "setLevel". think of a relayer we want to see what went wrong, without restarting

Reply:
1. I do not like the way named params are implemented in TypeScript so I avoid them when I can (<=3 parameters).
2. Clients should probably defautl to creating unique per-browser id, and only use userIdOverride when developing/testing
3. Changing log level at runtime is an advanced feature for now, not supported.